src/hera/workflows/artifact.py

```diff
@@ -97,6 +108,11 @@ def as_name(self, name: str) -> _ModelArtifact:
         artifact.name = name
         return artifact
 
+    @classmethod
+    def get_input_attributes(cls):
+        """Return the attributes used for input artifact annotations."""
+        return ["name", "path", "archive"]
```

Comment:
Archive would only be for outputs. More info on https://argoproj.github.io/argo-workflows/fields/#artifact